Clean up README.md (#128)
bufbuild/buf-gradle-plugin: README.md

[![Maven Central](https://img.shields.io/badge/dynamic/xml?color=orange&label=maven-central&prefix=v&query=%2F%2Fmetadata%2Fversioning%2Flatest&url=https%3A%2F%2Frepo1.maven.org%2Fmaven2%2Fbuild%2Fbuf%2Fbuf-gradle-plugin%2Fmaven-metadata.xml)](https://search.maven.org/artifact/build.buf/buf-gradle-plugin)
[![Gradle Portal](https://img.shields.io/maven-metadata/v/https/plugins.gradle.org/m2/build/buf/buf-gradle-plugin/maven-metadata.xml.svg?label=gradle-portal&color=yellowgreen)](https://plugins.gradle.org/plugin/build.buf)

This project provides support for integrating [Buf][buf] with Gradle. It also enables integration with the [Protobuf Gradle plugin][protobuf-gradle-plugin].

This plugin supports straightforward usage of `buf lint`, `buf format`, and `buf generate`, and a self-contained integration between `buf build` and `buf breaking`.

## Usage

This plugin assumes that Buf is configured for the project root with a configured `buf.work.yaml`. 

> You can check out how to get a Buf workspace set up through the [Buf docs][buf-docs].
> You are also able to use this plugin without specifying a `buf.work.yaml`. Without a specified yaml, the plugin will scan all top-level directories for protobuf sources.

Reviewer: ```suggestion
Instructions for setting up a Buf workspace can be found in the [Buf docs][buf-docs].
The plugin can also be used without specifying a `buf.work.yaml`. Without a specified yaml, the plugin will scan all top-level directories for Protobuf sources.
```

[![Maven Central](https://img.shields.io/badge/dynamic/xml?color=orange&label=maven-central&prefix=v&query=%2F%2Fmetadata%2Fversioning%2Flatest&url=https%3A%2F%2Frepo1.maven.org%2Fmaven2%2Fbuild%2Fbuf%2Fbuf-gradle-plugin%2Fmaven-metadata.xml)](https://search.maven.org/artifact/build.buf/buf-gradle-plugin)
[![Gradle Portal](https://img.shields.io/maven-metadata/v/https/plugins.gradle.org/m2/build/buf/buf-gradle-plugin/maven-metadata.xml.svg?label=gradle-portal&color=yellowgreen)](https://plugins.gradle.org/plugin/build.buf)

> Integration for [Buf][buf] with Gradle. Supports integration purely between Buf and Gradle or additionally with the [protobuf-gradle-plugin][protobuf-gradle-plugin].

Reviewer: ```suggestion
This project provides support for integrating [Buf][buf] with Gradle. It also enables integration with the [Protobuf Gradle plugin][protobuf-gradle-plugin].
```

```

> When applied the plugin creates tasks:

Reviewer: Can we add a link above to instructions for the line

> By default this plugin assumes that Buf is configured for the project root (with or without a workspace `buf.work.yaml`)

Author: Hm, you're suggesting that we have more detailed set up instructions for when the user does not have buf yamls?

Reviewer: I'm suggesting linking out to the instructions for setting up buf yaml files in case the reader is not aware of what they are or how to configure them. Don't think you need to duplicate information from those docs

Author: Gotcha

Author: Added a link to buf docs getting started in the description doc.

> - `bufFormatApply` applies Buf's formatter to Protobuf code

Reviewer: Maybe just link out to the Buf commands these invoke? (`bufFormatApply`: Invokes `buf format`)

Author: Let me do a mix of this. I think the intention was to have some description of what the tasks are doing rather than deferring to the Buf docs. I'll add links to the buf documentation for each of the mirrored buf cli commands and i'll call them out in the bullets

```

When applied the plugin creates tasks:
- `bufFormatApply` applies [`buf format`](https://buf.build/docs/format/style/)
- `bufFormatCheck` validates [`buf format`](https://buf.build/docs/format/style/)
- `bufLint` validates [`buf lint`](https://buf.build/docs/breaking/overview/)

> - `bufBreaking` checks Protobuf against a previous version for backwards-incompatible changes through [`buf breaking`](https://buf.build/docs/breaking/overview/)

Reviewer: ```suggestion
- `bufBreaking` checks Protobuf schemas against a previous version for backwards-incompatible changes through [`buf breaking`](https://buf.build/docs/breaking/overview/)
```

    toolVersion = <version>
}
```

## Contributing

We'd love your help making this plugin better!

> Extensive instructions for building the library and generator plugins locally,

Reviewer: Wrong

running tests, and contributing to the repository are available in our
[`CONTRIBUTING.md` guide](./.github/CONTRIBUTING.md). Please check it out
for details.

## Ecosystem

> * [connect-swift]: Swift clients for idiomatic gRPC & Connect RPC

Reviewer: Kotlin?

    toolVersion = <version>
}
```

## Contributing

We'd love your help making this plugin better!

Extensive instructions for building the plugin locally,
running tests, and contributing to the repository are available in our

> [`CONTRIBUTING.md` guide](./.github/CONTRIBUTING.md). Please check it out
> for details.

Reviewer: ```suggestion
[`CONTRIBUTING.md` guide](./.github/CONTRIBUTING.md).
```

## Ecosystem

> * [connect-Kotlin]: Kotlin clients for idiomatic gRPC & Connect RPC

Reviewer: ```suggestion
* [connect-kotlin]: Kotlin clients for idiomatic gRPC & Connect RPC
```

* [connect-web]: TypeScript clients for web browsers
* [connect-go]: Service handlers and clients for GoLang

> * [Buf Studio][buf-studio]: web UI for ad-hoc RPCs

Reviewer: ```suggestion
* [Buf Studio][buf-studio]: Web UI for ad-hoc RPCs
```

## Status

This project is in beta, and we may make a few changes as we gather feedback
from early adopters. Join us on [Slack][slack]!

## Legal

Offered under the [Apache 2 license][license].

[buf]: https://buf.build/
[buf-docs]: https://buf.build/docs/tutorials/getting-started-with-buf-cli/
[buf-studio]: https://studio.buf.build/
[connect-kotlin]: https://github.com/bufbuild/connect-kotlin
[connect-go]: https://github.com/bufbuild/connect-go
[connect-web]: https://www.npmjs.com/package/@bufbuild/connect-web
[license]: https://github.com/bufbuild/connect-go/blob/main/LICENSE
[protobuf-gradle-plugin]: https://github.com/google/protobuf-gradle-plugin
[slack]: https://buf.build/links/slack

> 

Reviewer: ```suggestion
```
